tools/seed_xor.py

```diff
@@ -0,0 +1,173 @@
+
+# -----------------------------------------------------------------------------
+# Formatted usage examples shown with `-h`
+# -----------------------------------------------------------------------------
+usage = f"""
+Examples:
+
+    # Split into 3 deterministic parts
+    python3 seed_xor.py split "<seed phrase>" 3 --deterministic
+
+    # Split into 4 random parts
+    python3 seed_xor.py split "<seed phrase>" 4
+
+    # Combine parts back
+    python3 seed_xor.py combine "<part A>" "<part B>" "<part C>"
+"""
+
+# -----------------------------------------------------------------------------
+# Banner printed at runtime
+# -----------------------------------------------------------------------------
+BANNER = """
+*******************************************************************************
+ SeedSigner XOR Seed Backup Tool
+
+ This tool splits a seed into multiple XOR backups and can recombine them to
+ recover the original seed. It relies on the BIP39 wordlist to convert between
+ seed phrases and binary entropy. FOR TESTING / EDUCATIONAL PURPOSES ONLY!
+*******************************************************************************
+"""
+
+def get_wordnumber(word, wordlist=WORDLIST__ENGLISH):
+    if word in wordlist:
+        index = wordlist.index(word)
+        return format(index, '011b') # 11-bit binary representation
+    return "Error"
+
+def mnemonic_to_entropy(phrase, wordlist=WORDLIST__ENGLISH):
+    mnemonic = ""
+    phrase = phrase.split()
+    for word in phrase:
+        mnemonic += get_wordnumber(word, wordlist)
+    
+    
+    checksum_bits = len(mnemonic) // 32 #remove checksum bits
+    return mnemonic[:-checksum_bits]
+
+def resize_bin(bin_str, nbits):
+    if nbits - len(bin_str) > 0:
+        bin_str = "0" * (nbits - len(bin_str)) + bin_str
+    return bin_str
+
+def entropy_to_mnemonic(entropy, wordlist=WORDLIST__ENGLISH):
+    #converts entropy to hex
+    if isinstance(entropy, int):
+        entropy_bin = bin(entropy)[2:]
+    else:
+        entropy_bin = entropy
+    
+    #it checks and adjust entropy size
+    entropy_sizes = [128, 160, 192, 224, 256]
+    entropy_size = next((size for size in entropy_sizes if len(entropy_bin) <= size), None)
+    
+    if not entropy_size:
+        return "Error: Invalid entropy size"
+    
+    entropy_bin = resize_bin(entropy_bin, entropy_size)
+    entropy_hex = hex(int(entropy_bin, 2))[2:]
+    entropy_hex = entropy_hex if len(entropy_hex) % 2 == 0 else "0" + entropy_hex
+    
+    #calculating checksum
+    entropy_bytes = a2b_hex(entropy_hex)
+    hash_bytes = sha256(entropy_bytes).digest()
+    checksum_bits = len(entropy_bin) // 32
+    checksum = ''.join(format(b, '08b') for b in hash_bytes)[:checksum_bits]
+    
+    full_binary = entropy_bin + checksum
+    
+    words = []
+    for i in range(0, len(full_binary), 11):
+        idx = int(full_binary[i:i+11], 2)
+        words.append(wordlist[idx])
+    
+    return " ".join(words)
```

Comment:
Isn't this redundant since we already use this elsewhere:

https://github.com/SeedSigner/seedsigner/blob/dev/src/seedsigner/helpers/mnemonic_generation.py#L59-L60